kernel/task.h

```diff
@@ -78,6 +88,8 @@ struct task {
     cond_t *waiting_cond;
     lock_t *waiting_lock;
     lock_t waiting_cond_lock;
+
+    bool traced;
```

Comment:
Would this make more sense inside of .ptrace?

Reply:
Yeah, I think it should be protected by the lock too.